packages/edit-site/src/components/left-sidebar/navigation-panel/index.js

```diff
@@ -1,80 +1,101 @@
 /**
  * WordPress dependencies
  */
-import { useEffect, useRef } from '@wordpress/element';
+import { useEffect, useRef, useState } from '@wordpress/element';
 import {
 	__experimentalNavigation as Navigation,
 	__experimentalNavigationMenu as NavigationMenu,
+	__experimentalNavigationItem as NavigationItem,
 	__experimentalNavigationBackButton as NavigationBackButton,
+	createSlotFill,
 } from '@wordpress/components';
 import { useDispatch, useSelect } from '@wordpress/data';
 import { __ } from '@wordpress/i18n';
 
 /**
  * Internal dependencies
  */
-import TemplateSwitcher from './template-switcher';
+import TemplatesMenu from './menus/templates';
+import TemplatePartsMenu from './menus/template-parts';
+
+export const {
+	Fill: NavigationPanelPreviewFill,
+	Slot: NavigationPanelPreviewSlot,
+} = createSlotFill( 'EditSiteNavigationPanelPreview' );
 
 const NavigationPanel = () => {
+	const [ activeMenu, setActiveMenu ] = useState( 'root' );
 	const ref = useRef();
 
 	useEffect( () => {
 		ref.current.focus();
 	}, [ ref ] );
 
-	const { templateId, templatePartId, templateType, page } = useSelect(
-		( select ) => {
-			const {
-				getTemplateId,
-				getTemplatePartId,
-				getTemplateType,
-				getPage,
-			} = select( 'core/edit-site' );
+	const { template, templatePart, templateType } = useSelect( ( select ) => {
+		const { getEntityRecord } = select( 'core' );
+		const { getTemplateId, getTemplatePartId, getTemplateType } = select(
+			'core/edit-site'
+		);
 
-			return {
-				templateId: getTemplateId(),
-				templatePartId: getTemplatePartId(),
-				templateType: getTemplateType(),
-				page: getPage(),
-			};
-		},
-		[]
-	);
+		const templateId = getTemplateId();
+		const templatePartId = getTemplatePartId();
+
+		return {
+			template: templateId
+				? getEntityRecord( 'postType', 'wp_template', templateId )
+				: null,
+			templatePart: templatePartId
+				? getEntityRecord(
+						'postType',
+						'wp_template_part',
+						templatePartId
+				  )
+				: null,
```

Comment:
What is the benefit of this change?
We are requesting the full entity records just to get the slug instead of the ID. Is the slug essential for the `activeItem`? 🤔

Reply:
Thanks for noticing. Reverted it to the ID version. I had to use slug because of a former solution but since then that's resolved.

Reply:
Reverted it back to the slug version 😬 ID changes after we created an auto-draft, which results in a brief moment when we don't have an active item in the navigation tree. Left a comment in the code to explain it.